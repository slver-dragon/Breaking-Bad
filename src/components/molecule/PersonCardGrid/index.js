import React from "react";
import { Link } from "react-router-dom";
import style from "./style.module.scss";

let statusStyle = "";

const PersonCard = ({ name, birthday, img, status, id }) => {
  switch (status) {
    case "Alive": {
      statusStyle = style.alive;
      break;
    }
    case "Deceased": {
      statusStyle = style.deceased;
      break;
    }
    default: {
      statusStyle = style.presumedDead;
    }
  }

  return (
    <Link to={`/person/${id}`} className={style.main}>
      <div className={style.photoWrapper}>
        <img className={style.photo} src={img} alt={name} />
      </div>
      <div className={style.inform}>
        <p className={style.state + " " + statusStyle}>{status}</p>
        <p className={style.name}>{name}</p>
        <p className={style.birthday}>{birthday}</p>
      </div>
    </Link>
  );
};

export default PersonCard;
