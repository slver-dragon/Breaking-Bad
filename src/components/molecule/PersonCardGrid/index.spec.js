import React from "react";
import { shallow } from "enzyme";
import PersonCardGrid from ".";

describe("Testing molecule Paginator:", () => {
  let component = null;
  const setComponent = (props) => shallow(<PersonCardGrid {...props} />);

  test("is rendering", () => {
    component = setComponent();
    expect(component).toMatchSnapshot();
  });

  test("status change", () => {
    component = setComponent({ status: "Alive" });
    expect(component.find(".alive").length).toBe(1);
    component = setComponent({ status: "Deceased" });
    expect(component.find(".deceased").length).toBe(1);
    component = setComponent({ status: "Presumed Dead" });
    expect(component.find(".presumedDead").length).toBe(1);
  });
});
