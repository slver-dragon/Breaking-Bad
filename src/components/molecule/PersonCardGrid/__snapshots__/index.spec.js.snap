// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing molecule Paginator: is rendering 1`] = `
<Link
  className="main"
  to="/person/undefined"
>
  <div
    className="photoWrapper"
  >
    <img
      className="photo"
    />
  </div>
  <div
    className="inform"
  >
    <p
      className="state presumedDead"
    />
    <p
      className="name"
    />
    <p
      className="birthday"
    />
  </div>
</Link>
`;
